tests/common/compare.py

```diff
@@ -53,8 +53,16 @@ def _check_item(item1: Any, item2: Any, path: str, rtol: float = 0.0, atol: floa
         # Increase update count so Torchmetrics doesn't throw warning when computing two metrics which haven't been updated
         item1._update_count += 1
         item2._update_count += 1
-        assert item1.compute().allclose(item2.compute(), atol=atol,
-                                        rtol=rtol), f'{path} differs: {item1.compute()} != {item2.compute()}'
+        item1_compute = item1.compute()
+        item2_compute = item2.compute()
+        if isinstance(item1_compute, torch.Tensor) and isinstance(item2_compute, torch.Tensor):
+            assert item1_compute.allclose(item2_compute, atol=atol, rtol=rtol,
+                                          equal_nan=True), f'{path} differs: {item1_compute} != {item2_compute}'
+        elif isinstance(item1_compute, dict):
+            assert isinstance(item2_compute, dict)
```

Comment:
this seems superfluous?

Reply:
the dict case? it's required for MAP

Reply:
no, the isinstance and the assert. is that just pyright being dumb?

Reply:
I am comparing 1 and 2. I am checking if 1 is a dict and requiring 2 to be a dict as well in that case

Reply:
ah, my b